### PR TITLE
Check for consistency in "add_neuron_meta"

### PR DESCRIPTION
**Pull request by kklurz**
The `add_neuron_meta` method in the FileTreeDataset does not check for consistency of the total number of neurons possible in the dataset and the actual number of neurons in the instantiation of the dataset. In the case of active subsampling transforms in the dataset instance, this can result in an error of inconsistent neuron indices later on which is very hard to debug.

### CODE REVIEW

File: neuralpredictors/data/datasets.py [hunk 1]
```diff
@@ -706,17 +680,18 @@ def add_neuron_meta(self, name, animal_id, session, scan_idx, unit_id, values, f
             values:     new meta information. First dimension must refer to neurons.
             fill_missing:   fill the values of the new attribute with NaN if not provided
         """
+        with HiddenPrints():
+            temp = NewFileTreeDataset(self.dirname, "responses")
```
Comment (fabiansinz) [inline neuralpredictors/data/datasets.py:684]: I would do `NewFileTreeDataset(self.dirname, *self.data_keys)`.

Comment (eywalker) [reply neuralpredictors/data/datasets.py:684]: Per discussion, we suggested that a new context manager is implemented to temporally turn off the transforms

Comment (mohammadbashiri) [reply neuralpredictors/data/datasets.py]: Thanks for the suggestion. It's implemented.

File: neuralpredictors/data/datasets.py [hunk 2]
```diff
@@ -706,28 +680,31 @@ def add_neuron_meta(self, name, animal_id, session, scan_idx, unit_id, values, f
...
-                self.neurons.scan_idx,
-                self.neurons.unit_ids,
+        with no_transforms(self):
+
+            if not self.n_neurons == len(values):
+                raise InconsistentDataException(
+                    f"Number of values ({len(values)}) and neurons in the datasets ({self.n_neurons}) is not consistent."
+                )
+
+            if not len(animal_id) == len(session) == len(scan_idx) == len(unit_id) == len(values):
```
Comment (fabiansinz) [inline neuralpredictors/data/datasets.py:691]: Unindent this!

Comment (kklurz) [reply neuralpredictors/data/datasets.py]: done

**Review by mohammadbashiri — APPROVED** [review]

**Pull request merged**